extensions/pyRevitTools.extension/pyRevit.tab/Toggles.panel/toggles2.stack/Imported.pushbutton/script.py

```diff
@@ -1,11 +1,26 @@
 """Toggles visibility of imported categories on current view."""
 
-from pyrevit import revit
+from pyrevit import revit, forms, DB
 
+def vg_parameter_controlled_by_template(view, built_in_parameter):
+    """
+    Checks if certain view parameter is overrided by template
+    """
+    if view.ViewTemplateId == DB.ElementId.InvalidElementId:
+        return False
+    view_template = view.Document.GetElement(view.ViewTemplateId)
+    if DB.ElementId(int(built_in_parameter)) in view_template.GetNonControlledTemplateParameterIds():
+        return False
+    else:
+        return True
 
 @revit.carryout('Toggle Imported')
 def toggle_imported():
     aview = revit.active_view
+    # check if 'Imported' is overrided by Template. If so, enable temporary view
```

Comment:
I find this issue pretty critical, actually. At the moment, Revit API allows to change `VIS_GRAPHICS_IMPORT` parameter of a view even if it was overwritten by View Template. After that there is almost no way to discover the reason of the issue. And the only way to change it back after that is to disable Template and set it back (I am gonna write to Autodesk about this issue).
Here script activates Temporary view settings if the parameter is overwritten by View Template

Reply:
Just for info, there is an Autodesk ticket with number `REVIT-153828` for the issue